{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "122d7cd6-a09d-4a0d-ae80-5c702e18c033",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Obtaining file:///notebooks/katlas\n",
      "  Preparing metadata (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25hRequirement already satisfied: fastcore in /usr/local/lib/python3.9/dist-packages (from katlas==0.0.1) (1.5.27)\n",
      "Requirement already satisfied: pandas in /usr/local/lib/python3.9/dist-packages (from katlas==0.0.1) (1.4.3)\n",
      "Collecting logomaker\n",
      "  Downloading logomaker-0.8-py2.py3-none-any.whl (11.8 MB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m11.8/11.8 MB\u001b[0m \u001b[31m54.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m00:01\u001b[0m00:01\u001b[0m\n",
      "\u001b[?25hRequirement already satisfied: seaborn in /usr/local/lib/python3.9/dist-packages (from katlas==0.0.1) (0.11.2)\n",
      "Collecting rdkit\n",
      "  Downloading rdkit-2023.3.3-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (29.7 MB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m29.7/29.7 MB\u001b[0m \u001b[31m53.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m00:01\u001b[0m00:01\u001b[0m\n",
      "\u001b[?25hCollecting fairscale\n",
      "  Downloading fairscale-0.4.13.tar.gz (266 kB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m266.3/266.3 kB\u001b[0m \u001b[31m52.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25h  Installing build dependencies ... \u001b[?25ldone\n",
      "\u001b[?25h  Getting requirements to build wheel ... \u001b[?25ldone\n",
      "\u001b[?25h  Installing backend dependencies ... \u001b[?25ldone\n",
      "\u001b[?25h  Preparing metadata (pyproject.toml) ... \u001b[?25ldone\n",
      "\u001b[?25hCollecting fair-esm\n",
      "  Downloading fair_esm-2.0.0-py3-none-any.whl (93 kB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m93.1/93.1 kB\u001b[0m \u001b[31m30.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hCollecting umap-learn\n",
      "  Downloading umap-learn-0.5.3.tar.gz (88 kB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m88.2/88.2 kB\u001b[0m \u001b[31m26.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25hCollecting adjustText\n",
      "  Downloading adjustText-0.8-py3-none-any.whl (9.1 kB)\n",
      "Requirement already satisfied: fastbook in /usr/local/lib/python3.9/dist-packages (from katlas==0.0.1) (0.0.28)\n",
      "Requirement already satisfied: xgboost in /usr/local/lib/python3.9/dist-packages (from katlas==0.0.1) (1.6.1)\n",
      "Collecting biopython\n",
      "  Downloading biopython-1.81-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.1 MB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m3.1/3.1 MB\u001b[0m \u001b[31m97.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hCollecting torchsummary\n",
      "  Downloading torchsummary-1.5.1-py3-none-any.whl (2.8 kB)\n",
      "Collecting nbdev\n",
      "  Downloading nbdev-2.3.12-py3-none-any.whl (64 kB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m64.8/64.8 kB\u001b[0m \u001b[31m21.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hRequirement already satisfied: numpy in /usr/local/lib/python3.9/dist-packages (from adjustText->katlas==0.0.1) (1.23.1)\n",
      "Requirement already satisfied: matplotlib in /usr/local/lib/python3.9/dist-packages (from adjustText->katlas==0.0.1) (3.5.2)\n",
      "Requirement already satisfied: torch>=1.8.0 in /usr/local/lib/python3.9/dist-packages (from fairscale->katlas==0.0.1) (1.12.0+cu116)\n",
      "Requirement already satisfied: sentencepiece in /usr/local/lib/python3.9/dist-packages (from fastbook->katlas==0.0.1) (0.1.96)\n",
      "Requirement already satisfied: fastai>=2.6 in /usr/local/lib/python3.9/dist-packages (from fastbook->katlas==0.0.1) (2.7.9)\n",
      "Requirement already satisfied: requests in /usr/local/lib/python3.9/dist-packages (from fastbook->katlas==0.0.1) (2.28.1)\n",
      "Requirement already satisfied: transformers in /usr/local/lib/python3.9/dist-packages (from fastbook->katlas==0.0.1) (4.20.1)\n",
      "Requirement already satisfied: datasets in /usr/local/lib/python3.9/dist-packages (from fastbook->katlas==0.0.1) (2.3.2)\n",
      "Requirement already satisfied: pip in /usr/local/lib/python3.9/dist-packages (from fastbook->katlas==0.0.1) (22.2.2)\n",
      "Requirement already satisfied: graphviz in /usr/local/lib/python3.9/dist-packages (from fastbook->katlas==0.0.1) (0.20.1)\n",
      "Requirement already satisfied: packaging in /usr/local/lib/python3.9/dist-packages (from fastbook->katlas==0.0.1) (21.3)\n",
      "Collecting execnb>=0.1.4\n",
      "  Downloading execnb-0.1.5-py3-none-any.whl (13 kB)\n",
      "Requirement already satisfied: PyYAML in /usr/lib/python3/dist-packages (from nbdev->katlas==0.0.1) (5.3.1)\n",
      "Requirement already satisfied: asttokens in /usr/local/lib/python3.9/dist-packages (from nbdev->katlas==0.0.1) (2.0.5)\n",
      "Collecting ghapi>=1.0.3\n",
      "  Downloading ghapi-1.0.4-py3-none-any.whl (58 kB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m58.7/58.7 kB\u001b[0m \u001b[31m20.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hCollecting watchdog\n",
      "  Downloading watchdog-3.0.0-py3-none-manylinux2014_x86_64.whl (82 kB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m82.1/82.1 kB\u001b[0m \u001b[31m28.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hRequirement already satisfied: astunparse in /usr/local/lib/python3.9/dist-packages (from nbdev->katlas==0.0.1) (1.6.3)\n",
      "Requirement already satisfied: python-dateutil>=2.8.1 in /usr/local/lib/python3.9/dist-packages (from pandas->katlas==0.0.1) (2.8.2)\n",
      "Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.9/dist-packages (from pandas->katlas==0.0.1) (2022.1)\n",
      "Requirement already satisfied: Pillow in /usr/local/lib/python3.9/dist-packages (from rdkit->katlas==0.0.1) (9.2.0)\n",
      "Requirement already satisfied: scipy>=1.0 in /usr/local/lib/python3.9/dist-packages (from seaborn->katlas==0.0.1) (1.8.1)\n",
      "Requirement already satisfied: scikit-learn>=0.22 in /usr/local/lib/python3.9/dist-packages (from umap-learn->katlas==0.0.1) (1.1.1)\n",
      "Collecting numba>=0.49\n",
      "  Downloading numba-0.57.1-cp39-cp39-manylinux2014_x86_64.manylinux_2_17_x86_64.whl (3.6 MB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m3.6/3.6 MB\u001b[0m \u001b[31m119.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hCollecting pynndescent>=0.5\n",
      "  Downloading pynndescent-0.5.10.tar.gz (1.1 MB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.1/1.1 MB\u001b[0m \u001b[31m107.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25hRequirement already satisfied: tqdm in /usr/local/lib/python3.9/dist-packages (from umap-learn->katlas==0.0.1) (4.64.0)\n",
      "Requirement already satisfied: ipython in /usr/local/lib/python3.9/dist-packages (from execnb>=0.1.4->nbdev->katlas==0.0.1) (8.4.0)\n",
      "Requirement already satisfied: fastdownload<2,>=0.0.5 in /usr/local/lib/python3.9/dist-packages (from fastai>=2.6->fastbook->katlas==0.0.1) (0.0.7)\n",
      "Requirement already satisfied: fastprogress>=0.2.4 in /usr/local/lib/python3.9/dist-packages (from fastai>=2.6->fastbook->katlas==0.0.1) (1.0.3)\n",
      "Requirement already satisfied: spacy<4 in /usr/local/lib/python3.9/dist-packages (from fastai>=2.6->fastbook->katlas==0.0.1) (3.4.0)\n",
      "Requirement already satisfied: torchvision>=0.8.2 in /usr/local/lib/python3.9/dist-packages (from fastai>=2.6->fastbook->katlas==0.0.1) (0.13.0+cu116)\n",
      "Requirement already satisfied: pyparsing>=2.2.1 in /usr/local/lib/python3.9/dist-packages (from matplotlib->adjustText->katlas==0.0.1) (3.0.9)\n",
      "Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.9/dist-packages (from matplotlib->adjustText->katlas==0.0.1) (0.11.0)\n",
      "Requirement already satisfied: fonttools>=4.22.0 in /usr/local/lib/python3.9/dist-packages (from matplotlib->adjustText->katlas==0.0.1) (4.34.4)\n",
      "Requirement already satisfied: kiwisolver>=1.0.1 in /usr/local/lib/python3.9/dist-packages (from matplotlib->adjustText->katlas==0.0.1) (1.4.3)\n",
      "Collecting llvmlite<0.41,>=0.40.0dev0\n",
      "  Downloading llvmlite-0.40.1-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (42.1 MB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m42.1/42.1 MB\u001b[0m \u001b[31m43.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m00:01\u001b[0m00:01\u001b[0m\n",
      "\u001b[?25hRequirement already satisfied: joblib>=0.11 in /usr/local/lib/python3.9/dist-packages (from pynndescent>=0.5->umap-learn->katlas==0.0.1) (1.1.0)\n",
      "Requirement already satisfied: six>=1.5 in /usr/lib/python3/dist-packages (from python-dateutil>=2.8.1->pandas->katlas==0.0.1) (1.14.0)\n",
      "Requirement already satisfied: threadpoolctl>=2.0.0 in /usr/local/lib/python3.9/dist-packages (from scikit-learn>=0.22->umap-learn->katlas==0.0.1) (3.1.0)\n",
      "Requirement already satisfied: typing-extensions in /usr/local/lib/python3.9/dist-packages (from torch>=1.8.0->fairscale->katlas==0.0.1) (4.3.0)\n",
      "Requirement already satisfied: wheel<1.0,>=0.23.0 in /usr/local/lib/python3.9/dist-packages (from astunparse->nbdev->katlas==0.0.1) (0.35.1)\n",
      "Requirement already satisfied: xxhash in /usr/local/lib/python3.9/dist-packages (from datasets->fastbook->katlas==0.0.1) (3.0.0)\n",
      "Requirement already satisfied: aiohttp in /usr/local/lib/python3.9/dist-packages (from datasets->fastbook->katlas==0.0.1) (3.8.1)\n",
      "Requirement already satisfied: pyarrow>=6.0.0 in /usr/local/lib/python3.9/dist-packages (from datasets->fastbook->katlas==0.0.1) (8.0.0)\n",
      "Requirement already satisfied: huggingface-hub<1.0.0,>=0.1.0 in /usr/local/lib/python3.9/dist-packages (from datasets->fastbook->katlas==0.0.1) (0.8.1)\n",
      "Requirement already satisfied: responses<0.19 in /usr/local/lib/python3.9/dist-packages (from datasets->fastbook->katlas==0.0.1) (0.18.0)\n",
      "Requirement already satisfied: multiprocess in /usr/local/lib/python3.9/dist-packages (from datasets->fastbook->katlas==0.0.1) (0.70.13)\n",
      "Requirement already satisfied: dill<0.3.6 in /usr/local/lib/python3.9/dist-packages (from datasets->fastbook->katlas==0.0.1) (0.3.5.1)\n",
      "Requirement already satisfied: fsspec[http]>=2021.05.0 in /usr/local/lib/python3.9/dist-packages (from datasets->fastbook->katlas==0.0.1) (2022.5.0)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /usr/lib/python3/dist-packages (from requests->fastbook->katlas==0.0.1) (2019.11.28)\n",
      "Requirement already satisfied: charset-normalizer<3,>=2 in /usr/local/lib/python3.9/dist-packages (from requests->fastbook->katlas==0.0.1) (2.1.0)\n",
      "Requirement already satisfied: urllib3<1.27,>=1.21.1 in /usr/local/lib/python3.9/dist-packages (from requests->fastbook->katlas==0.0.1) (1.26.10)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /usr/lib/python3/dist-packages (from requests->fastbook->katlas==0.0.1) (2.8)\n",
      "Requirement already satisfied: filelock in /usr/local/lib/python3.9/dist-packages (from transformers->fastbook->katlas==0.0.1) (3.7.1)\n",
      "Requirement already satisfied: tokenizers!=0.11.3,<0.13,>=0.11.1 in /usr/local/lib/python3.9/dist-packages (from transformers->fastbook->katlas==0.0.1) (0.12.1)\n",
      "Requirement already satisfied: regex!=2019.12.17 in /usr/local/lib/python3.9/dist-packages (from transformers->fastbook->katlas==0.0.1) (2022.7.9)\n",
      "Requirement already satisfied: spacy-loggers<2.0.0,>=1.0.0 in /usr/local/lib/python3.9/dist-packages (from spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (1.0.2)\n",
      "Requirement already satisfied: catalogue<2.1.0,>=2.0.6 in /usr/local/lib/python3.9/dist-packages (from spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (2.0.7)\n",
      "Requirement already satisfied: thinc<8.2.0,>=8.1.0 in /usr/local/lib/python3.9/dist-packages (from spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (8.1.0)\n",
      "Requirement already satisfied: spacy-legacy<3.1.0,>=3.0.9 in /usr/local/lib/python3.9/dist-packages (from spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (3.0.9)\n",
      "Requirement already satisfied: jinja2 in /usr/local/lib/python3.9/dist-packages (from spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (3.1.2)\n",
      "Requirement already satisfied: pydantic!=1.8,!=1.8.1,<1.10.0,>=1.7.4 in /usr/local/lib/python3.9/dist-packages (from spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (1.9.1)\n",
      "Requirement already satisfied: setuptools in /usr/local/lib/python3.9/dist-packages (from spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (63.1.0)\n",
      "Requirement already satisfied: cymem<2.1.0,>=2.0.2 in /usr/local/lib/python3.9/dist-packages (from spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (2.0.6)\n",
      "Requirement already satisfied: murmurhash<1.1.0,>=0.28.0 in /usr/local/lib/python3.9/dist-packages (from spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (1.0.7)\n",
      "Requirement already satisfied: typer<0.5.0,>=0.3.0 in /usr/local/lib/python3.9/dist-packages (from spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (0.4.2)\n",
      "Requirement already satisfied: preshed<3.1.0,>=3.0.2 in /usr/local/lib/python3.9/dist-packages (from spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (3.0.6)\n",
      "Requirement already satisfied: pathy>=0.3.5 in /usr/local/lib/python3.9/dist-packages (from spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (0.6.2)\n",
      "Requirement already satisfied: wasabi<1.1.0,>=0.9.1 in /usr/local/lib/python3.9/dist-packages (from spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (0.9.1)\n",
      "Requirement already satisfied: langcodes<4.0.0,>=3.2.0 in /usr/local/lib/python3.9/dist-packages (from spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (3.3.0)\n",
      "Requirement already satisfied: srsly<3.0.0,>=2.4.3 in /usr/local/lib/python3.9/dist-packages (from spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (2.4.3)\n",
      "Requirement already satisfied: yarl<2.0,>=1.0 in /usr/local/lib/python3.9/dist-packages (from aiohttp->datasets->fastbook->katlas==0.0.1) (1.7.2)\n",
      "Requirement already satisfied: async-timeout<5.0,>=4.0.0a3 in /usr/local/lib/python3.9/dist-packages (from aiohttp->datasets->fastbook->katlas==0.0.1) (4.0.2)\n",
      "Requirement already satisfied: frozenlist>=1.1.1 in /usr/local/lib/python3.9/dist-packages (from aiohttp->datasets->fastbook->katlas==0.0.1) (1.3.0)\n",
      "Requirement already satisfied: multidict<7.0,>=4.5 in /usr/local/lib/python3.9/dist-packages (from aiohttp->datasets->fastbook->katlas==0.0.1) (6.0.2)\n",
      "Requirement already satisfied: aiosignal>=1.1.2 in /usr/local/lib/python3.9/dist-packages (from aiohttp->datasets->fastbook->katlas==0.0.1) (1.2.0)\n",
      "Requirement already satisfied: attrs>=17.3.0 in /usr/local/lib/python3.9/dist-packages (from aiohttp->datasets->fastbook->katlas==0.0.1) (18.2.0)\n",
      "Requirement already satisfied: pygments>=2.4.0 in /usr/local/lib/python3.9/dist-packages (from ipython->execnb>=0.1.4->nbdev->katlas==0.0.1) (2.12.0)\n",
      "Requirement already satisfied: traitlets>=5 in /usr/local/lib/python3.9/dist-packages (from ipython->execnb>=0.1.4->nbdev->katlas==0.0.1) (5.3.0)\n",
      "Requirement already satisfied: pexpect>4.3 in /usr/local/lib/python3.9/dist-packages (from ipython->execnb>=0.1.4->nbdev->katlas==0.0.1) (4.8.0)\n",
      "Requirement already satisfied: pickleshare in /usr/local/lib/python3.9/dist-packages (from ipython->execnb>=0.1.4->nbdev->katlas==0.0.1) (0.7.5)\n",
      "Requirement already satisfied: prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0 in /usr/local/lib/python3.9/dist-packages (from ipython->execnb>=0.1.4->nbdev->katlas==0.0.1) (3.0.30)\n",
      "Requirement already satisfied: decorator in /usr/local/lib/python3.9/dist-packages (from ipython->execnb>=0.1.4->nbdev->katlas==0.0.1) (5.1.1)\n",
      "Requirement already satisfied: backcall in /usr/local/lib/python3.9/dist-packages (from ipython->execnb>=0.1.4->nbdev->katlas==0.0.1) (0.2.0)\n",
      "Requirement already satisfied: matplotlib-inline in /usr/local/lib/python3.9/dist-packages (from ipython->execnb>=0.1.4->nbdev->katlas==0.0.1) (0.1.3)\n",
      "Requirement already satisfied: jedi>=0.16 in /usr/local/lib/python3.9/dist-packages (from ipython->execnb>=0.1.4->nbdev->katlas==0.0.1) (0.18.1)\n",
      "Requirement already satisfied: stack-data in /usr/local/lib/python3.9/dist-packages (from ipython->execnb>=0.1.4->nbdev->katlas==0.0.1) (0.3.0)\n",
      "Requirement already satisfied: parso<0.9.0,>=0.8.0 in /usr/local/lib/python3.9/dist-packages (from jedi>=0.16->ipython->execnb>=0.1.4->nbdev->katlas==0.0.1) (0.8.3)\n",
      "Requirement already satisfied: smart-open<6.0.0,>=5.2.1 in /usr/local/lib/python3.9/dist-packages (from pathy>=0.3.5->spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (5.2.1)\n",
      "Requirement already satisfied: ptyprocess>=0.5 in /usr/local/lib/python3.9/dist-packages (from pexpect>4.3->ipython->execnb>=0.1.4->nbdev->katlas==0.0.1) (0.7.0)\n",
      "Requirement already satisfied: wcwidth in /usr/local/lib/python3.9/dist-packages (from prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0->ipython->execnb>=0.1.4->nbdev->katlas==0.0.1) (0.2.5)\n",
      "Requirement already satisfied: blis<0.8.0,>=0.7.8 in /usr/local/lib/python3.9/dist-packages (from thinc<8.2.0,>=8.1.0->spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (0.7.8)\n",
      "Requirement already satisfied: click<9.0.0,>=7.1.1 in /usr/local/lib/python3.9/dist-packages (from typer<0.5.0,>=0.3.0->spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (8.1.3)\n",
      "Requirement already satisfied: MarkupSafe>=2.0 in /usr/local/lib/python3.9/dist-packages (from jinja2->spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (2.1.1)\n",
      "Requirement already satisfied: pure-eval in /usr/local/lib/python3.9/dist-packages (from stack-data->ipython->execnb>=0.1.4->nbdev->katlas==0.0.1) (0.2.2)\n",
      "Requirement already satisfied: executing in /usr/local/lib/python3.9/dist-packages (from stack-data->ipython->execnb>=0.1.4->nbdev->katlas==0.0.1) (0.8.3)\n",
      "Building wheels for collected packages: fairscale, umap-learn, pynndescent\n",
      "  Building wheel for fairscale (pyproject.toml) ... \u001b[?25ldone\n",
      "\u001b[?25h  Created wheel for fairscale: filename=fairscale-0.4.13-py3-none-any.whl size=332113 sha256=146adf46317e1b212a87274b613ebcff32d0cafef1af6f02372d7104b08fab0d\n",
      "  Stored in directory: /root/.cache/pip/wheels/10/ea/7f/8f35af83599829bb4790bdc16949dd99aeeb62e9a1faf47d47\n",
      "  Building wheel for umap-learn (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25h  Created wheel for umap-learn: filename=umap_learn-0.5.3-py3-none-any.whl size=82813 sha256=e4b99667032e6ff53702da15de20977a9ff27df978794b86c762e71830a0f9b7\n",
      "  Stored in directory: /root/.cache/pip/wheels/f4/3e/1c/596d0a463d17475af648688443fa4846fef624d1390339e7e9\n",
      "  Building wheel for pynndescent (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25h  Created wheel for pynndescent: filename=pynndescent-0.5.10-py3-none-any.whl size=55622 sha256=bf1b364e7237b4d6eca0736766f0890eba9baac75157055271ec57e64f199868\n",
      "  Stored in directory: /root/.cache/pip/wheels/12/f9/4d/ec5ad1c823c710fcc4473669fdcffc8891f4bc398c841af22e\n",
      "Successfully built fairscale umap-learn pynndescent\n",
      "Installing collected packages: torchsummary, fair-esm, watchdog, rdkit, llvmlite, biopython, numba, fairscale, pynndescent, logomaker, ghapi, adjustText, umap-learn, execnb, nbdev, katlas\n",
      "  Running setup.py develop for katlas\n",
      "Successfully installed adjustText-0.8 biopython-1.81 execnb-0.1.5 fair-esm-2.0.0 fairscale-0.4.13 ghapi-1.0.4 katlas llvmlite-0.40.1 logomaker-0.8 nbdev-2.3.12 numba-0.57.1 pynndescent-0.5.10 rdkit-2023.3.3 torchsummary-1.5.1 umap-learn-0.5.3 watchdog-3.0.0\n",
      "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n",
      "\u001b[0mInstalling or upgrading quarto -- this requires root access.\n",
      "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
      "                                 Dload  Upload   Total   Spent    Left  Speed\n",
      "100    72  100    72    0     0    600      0 --:--:-- --:--:-- --:--:--   600\n",
      "100   114  100   114    0     0    430      0 --:--:-- --:--:-- --:--:--   430\n",
      "  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\n",
      "100 84.5M  100 84.5M    0     0   110M      0 --:--:-- --:--:-- --:--:--  110M\n",
      "Selecting previously unselected package quarto.\n",
      "(Reading database ... 78769 files and directories currently installed.)\n",
      "Preparing to unpack quarto-linux-amd64.deb ...\n",
      "Unpacking quarto (1.3.450) ...\n",
      "Setting up quarto (1.3.450) ...\n"
     ]
    }
   ],
   "source": [
    "!pip install -e '.[dev]'\n",
    "\n",
    "!nbdev_install_quarto\n",
    "!pip install -Uqq scikit-learn\n",
    "!pip install -Uqq fastai\n",
    "# !pip install cuml-cu11 --extra-index-url=https://pypi.nvidia.com\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "1eda98a7-51d8-40c8-a71d-3ee1f037ce61",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing /notebooks/katlas\n",
      "  Preparing metadata (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25hRequirement already satisfied: fastcore in /usr/local/lib/python3.9/dist-packages (from katlas==0.0.1) (1.5.29)\n",
      "Requirement already satisfied: pandas in /usr/local/lib/python3.9/dist-packages (from katlas==0.0.1) (1.4.3)\n",
      "Requirement already satisfied: logomaker in /usr/local/lib/python3.9/dist-packages (from katlas==0.0.1) (0.8)\n",
      "Requirement already satisfied: seaborn in /usr/local/lib/python3.9/dist-packages (from katlas==0.0.1) (0.11.2)\n",
      "Requirement already satisfied: rdkit in /usr/local/lib/python3.9/dist-packages (from katlas==0.0.1) (2023.3.3)\n",
      "Requirement already satisfied: fairscale in /usr/local/lib/python3.9/dist-packages (from katlas==0.0.1) (0.4.13)\n",
      "Requirement already satisfied: fair-esm in /usr/local/lib/python3.9/dist-packages (from katlas==0.0.1) (2.0.0)\n",
      "Requirement already satisfied: umap-learn in /usr/local/lib/python3.9/dist-packages (from katlas==0.0.1) (0.5.3)\n",
      "Requirement already satisfied: adjustText in /usr/local/lib/python3.9/dist-packages (from katlas==0.0.1) (0.8)\n",
      "Requirement already satisfied: fastbook in /usr/local/lib/python3.9/dist-packages (from katlas==0.0.1) (0.0.28)\n",
      "Requirement already satisfied: xgboost in /usr/local/lib/python3.9/dist-packages (from katlas==0.0.1) (1.6.1)\n",
      "Requirement already satisfied: biopython in /usr/local/lib/python3.9/dist-packages (from katlas==0.0.1) (1.81)\n",
      "Requirement already satisfied: torchsummary in /usr/local/lib/python3.9/dist-packages (from katlas==0.0.1) (1.5.1)\n",
      "Requirement already satisfied: matplotlib in /usr/local/lib/python3.9/dist-packages (from adjustText->katlas==0.0.1) (3.5.2)\n",
      "Requirement already satisfied: numpy in /usr/local/lib/python3.9/dist-packages (from adjustText->katlas==0.0.1) (1.23.1)\n",
      "Requirement already satisfied: torch>=1.8.0 in /usr/local/lib/python3.9/dist-packages (from fairscale->katlas==0.0.1) (1.12.0+cu116)\n",
      "Requirement already satisfied: pip in /usr/local/lib/python3.9/dist-packages (from fastbook->katlas==0.0.1) (22.2.2)\n",
      "Requirement already satisfied: transformers in /usr/local/lib/python3.9/dist-packages (from fastbook->katlas==0.0.1) (4.20.1)\n",
      "Requirement already satisfied: graphviz in /usr/local/lib/python3.9/dist-packages (from fastbook->katlas==0.0.1) (0.20.1)\n",
      "Requirement already satisfied: sentencepiece in /usr/local/lib/python3.9/dist-packages (from fastbook->katlas==0.0.1) (0.1.96)\n",
      "Requirement already satisfied: datasets in /usr/local/lib/python3.9/dist-packages (from fastbook->katlas==0.0.1) (2.3.2)\n",
      "Requirement already satisfied: requests in /usr/local/lib/python3.9/dist-packages (from fastbook->katlas==0.0.1) (2.28.1)\n",
      "Requirement already satisfied: packaging in /usr/local/lib/python3.9/dist-packages (from fastbook->katlas==0.0.1) (21.3)\n",
      "Requirement already satisfied: fastai>=2.6 in /usr/local/lib/python3.9/dist-packages (from fastbook->katlas==0.0.1) (2.7.12)\n",
      "Requirement already satisfied: python-dateutil>=2.8.1 in /usr/local/lib/python3.9/dist-packages (from pandas->katlas==0.0.1) (2.8.2)\n",
      "Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.9/dist-packages (from pandas->katlas==0.0.1) (2022.1)\n",
      "Requirement already satisfied: Pillow in /usr/local/lib/python3.9/dist-packages (from rdkit->katlas==0.0.1) (9.2.0)\n",
      "Requirement already satisfied: scipy>=1.0 in /usr/local/lib/python3.9/dist-packages (from seaborn->katlas==0.0.1) (1.8.1)\n",
      "Requirement already satisfied: scikit-learn>=0.22 in /usr/local/lib/python3.9/dist-packages (from umap-learn->katlas==0.0.1) (1.3.0)\n",
      "Requirement already satisfied: tqdm in /usr/local/lib/python3.9/dist-packages (from umap-learn->katlas==0.0.1) (4.64.0)\n",
      "Requirement already satisfied: numba>=0.49 in /usr/local/lib/python3.9/dist-packages (from umap-learn->katlas==0.0.1) (0.57.1)\n",
      "Requirement already satisfied: pynndescent>=0.5 in /usr/local/lib/python3.9/dist-packages (from umap-learn->katlas==0.0.1) (0.5.10)\n",
      "Requirement already satisfied: spacy<4 in /usr/local/lib/python3.9/dist-packages (from fastai>=2.6->fastbook->katlas==0.0.1) (3.4.0)\n",
      "Requirement already satisfied: fastdownload<2,>=0.0.5 in /usr/local/lib/python3.9/dist-packages (from fastai>=2.6->fastbook->katlas==0.0.1) (0.0.7)\n",
      "Requirement already satisfied: fastprogress>=0.2.4 in /usr/local/lib/python3.9/dist-packages (from fastai>=2.6->fastbook->katlas==0.0.1) (1.0.3)\n",
      "Requirement already satisfied: torchvision>=0.8.2 in /usr/local/lib/python3.9/dist-packages (from fastai>=2.6->fastbook->katlas==0.0.1) (0.13.0+cu116)\n",
      "Requirement already satisfied: pyyaml in /usr/lib/python3/dist-packages (from fastai>=2.6->fastbook->katlas==0.0.1) (5.3.1)\n",
      "Requirement already satisfied: fonttools>=4.22.0 in /usr/local/lib/python3.9/dist-packages (from matplotlib->adjustText->katlas==0.0.1) (4.34.4)\n",
      "Requirement already satisfied: pyparsing>=2.2.1 in /usr/local/lib/python3.9/dist-packages (from matplotlib->adjustText->katlas==0.0.1) (3.0.9)\n",
      "Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.9/dist-packages (from matplotlib->adjustText->katlas==0.0.1) (0.11.0)\n",
      "Requirement already satisfied: kiwisolver>=1.0.1 in /usr/local/lib/python3.9/dist-packages (from matplotlib->adjustText->katlas==0.0.1) (1.4.3)\n",
      "Requirement already satisfied: llvmlite<0.41,>=0.40.0dev0 in /usr/local/lib/python3.9/dist-packages (from numba>=0.49->umap-learn->katlas==0.0.1) (0.40.1)\n",
      "Requirement already satisfied: joblib>=0.11 in /usr/local/lib/python3.9/dist-packages (from pynndescent>=0.5->umap-learn->katlas==0.0.1) (1.3.2)\n",
      "Requirement already satisfied: six>=1.5 in /usr/lib/python3/dist-packages (from python-dateutil>=2.8.1->pandas->katlas==0.0.1) (1.14.0)\n",
      "Requirement already satisfied: threadpoolctl>=2.0.0 in /usr/local/lib/python3.9/dist-packages (from scikit-learn>=0.22->umap-learn->katlas==0.0.1) (3.1.0)\n",
      "Requirement already satisfied: typing-extensions in /usr/local/lib/python3.9/dist-packages (from torch>=1.8.0->fairscale->katlas==0.0.1) (4.3.0)\n",
      "Requirement already satisfied: aiohttp in /usr/local/lib/python3.9/dist-packages (from datasets->fastbook->katlas==0.0.1) (3.8.1)\n",
      "Requirement already satisfied: huggingface-hub<1.0.0,>=0.1.0 in /usr/local/lib/python3.9/dist-packages (from datasets->fastbook->katlas==0.0.1) (0.8.1)\n",
      "Requirement already satisfied: xxhash in /usr/local/lib/python3.9/dist-packages (from datasets->fastbook->katlas==0.0.1) (3.0.0)\n",
      "Requirement already satisfied: fsspec[http]>=2021.05.0 in /usr/local/lib/python3.9/dist-packages (from datasets->fastbook->katlas==0.0.1) (2022.5.0)\n",
      "Requirement already satisfied: responses<0.19 in /usr/local/lib/python3.9/dist-packages (from datasets->fastbook->katlas==0.0.1) (0.18.0)\n",
      "Requirement already satisfied: dill<0.3.6 in /usr/local/lib/python3.9/dist-packages (from datasets->fastbook->katlas==0.0.1) (0.3.5.1)\n",
      "Requirement already satisfied: pyarrow>=6.0.0 in /usr/local/lib/python3.9/dist-packages (from datasets->fastbook->katlas==0.0.1) (8.0.0)\n",
      "Requirement already satisfied: multiprocess in /usr/local/lib/python3.9/dist-packages (from datasets->fastbook->katlas==0.0.1) (0.70.13)\n",
      "Requirement already satisfied: charset-normalizer<3,>=2 in /usr/local/lib/python3.9/dist-packages (from requests->fastbook->katlas==0.0.1) (2.1.0)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /usr/lib/python3/dist-packages (from requests->fastbook->katlas==0.0.1) (2019.11.28)\n",
      "Requirement already satisfied: urllib3<1.27,>=1.21.1 in /usr/local/lib/python3.9/dist-packages (from requests->fastbook->katlas==0.0.1) (1.26.10)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /usr/lib/python3/dist-packages (from requests->fastbook->katlas==0.0.1) (2.8)\n",
      "Requirement already satisfied: filelock in /usr/local/lib/python3.9/dist-packages (from transformers->fastbook->katlas==0.0.1) (3.7.1)\n",
      "Requirement already satisfied: regex!=2019.12.17 in /usr/local/lib/python3.9/dist-packages (from transformers->fastbook->katlas==0.0.1) (2022.7.9)\n",
      "Requirement already satisfied: tokenizers!=0.11.3,<0.13,>=0.11.1 in /usr/local/lib/python3.9/dist-packages (from transformers->fastbook->katlas==0.0.1) (0.12.1)\n",
      "Requirement already satisfied: pathy>=0.3.5 in /usr/local/lib/python3.9/dist-packages (from spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (0.6.2)\n",
      "Requirement already satisfied: setuptools in /usr/local/lib/python3.9/dist-packages (from spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (63.1.0)\n",
      "Requirement already satisfied: typer<0.5.0,>=0.3.0 in /usr/local/lib/python3.9/dist-packages (from spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (0.4.2)\n",
      "Requirement already satisfied: spacy-loggers<2.0.0,>=1.0.0 in /usr/local/lib/python3.9/dist-packages (from spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (1.0.2)\n",
      "Requirement already satisfied: thinc<8.2.0,>=8.1.0 in /usr/local/lib/python3.9/dist-packages (from spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (8.1.0)\n",
      "Requirement already satisfied: catalogue<2.1.0,>=2.0.6 in /usr/local/lib/python3.9/dist-packages (from spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (2.0.7)\n",
      "Requirement already satisfied: srsly<3.0.0,>=2.4.3 in /usr/local/lib/python3.9/dist-packages (from spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (2.4.3)\n",
      "Requirement already satisfied: spacy-legacy<3.1.0,>=3.0.9 in /usr/local/lib/python3.9/dist-packages (from spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (3.0.9)\n",
      "Requirement already satisfied: pydantic!=1.8,!=1.8.1,<1.10.0,>=1.7.4 in /usr/local/lib/python3.9/dist-packages (from spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (1.9.1)\n",
      "Requirement already satisfied: cymem<2.1.0,>=2.0.2 in /usr/local/lib/python3.9/dist-packages (from spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (2.0.6)\n",
      "Requirement already satisfied: murmurhash<1.1.0,>=0.28.0 in /usr/local/lib/python3.9/dist-packages (from spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (1.0.7)\n",
      "Requirement already satisfied: wasabi<1.1.0,>=0.9.1 in /usr/local/lib/python3.9/dist-packages (from spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (0.9.1)\n",
      "Requirement already satisfied: jinja2 in /usr/local/lib/python3.9/dist-packages (from spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (3.1.2)\n",
      "Requirement already satisfied: langcodes<4.0.0,>=3.2.0 in /usr/local/lib/python3.9/dist-packages (from spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (3.3.0)\n",
      "Requirement already satisfied: preshed<3.1.0,>=3.0.2 in /usr/local/lib/python3.9/dist-packages (from spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (3.0.6)\n",
      "Requirement already satisfied: yarl<2.0,>=1.0 in /usr/local/lib/python3.9/dist-packages (from aiohttp->datasets->fastbook->katlas==0.0.1) (1.7.2)\n",
      "Requirement already satisfied: async-timeout<5.0,>=4.0.0a3 in /usr/local/lib/python3.9/dist-packages (from aiohttp->datasets->fastbook->katlas==0.0.1) (4.0.2)\n",
      "Requirement already satisfied: aiosignal>=1.1.2 in /usr/local/lib/python3.9/dist-packages (from aiohttp->datasets->fastbook->katlas==0.0.1) (1.2.0)\n",
      "Requirement already satisfied: frozenlist>=1.1.1 in /usr/local/lib/python3.9/dist-packages (from aiohttp->datasets->fastbook->katlas==0.0.1) (1.3.0)\n",
      "Requirement already satisfied: attrs>=17.3.0 in /usr/local/lib/python3.9/dist-packages (from aiohttp->datasets->fastbook->katlas==0.0.1) (18.2.0)\n",
      "Requirement already satisfied: multidict<7.0,>=4.5 in /usr/local/lib/python3.9/dist-packages (from aiohttp->datasets->fastbook->katlas==0.0.1) (6.0.2)\n",
      "Requirement already satisfied: smart-open<6.0.0,>=5.2.1 in /usr/local/lib/python3.9/dist-packages (from pathy>=0.3.5->spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (5.2.1)\n",
      "Requirement already satisfied: blis<0.8.0,>=0.7.8 in /usr/local/lib/python3.9/dist-packages (from thinc<8.2.0,>=8.1.0->spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (0.7.8)\n",
      "Requirement already satisfied: click<9.0.0,>=7.1.1 in /usr/local/lib/python3.9/dist-packages (from typer<0.5.0,>=0.3.0->spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (8.1.3)\n",
      "Requirement already satisfied: MarkupSafe>=2.0 in /usr/local/lib/python3.9/dist-packages (from jinja2->spacy<4->fastai>=2.6->fastbook->katlas==0.0.1) (2.1.1)\n",
      "Building wheels for collected packages: katlas\n",
      "  Building wheel for katlas (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25h  Created wheel for katlas: filename=katlas-0.0.1-py3-none-any.whl size=27765 sha256=696d3be68e594eb1d2dfae00be3859c6e68dc35d45a810214d8dbfebd78a53c8\n",
      "  Stored in directory: /tmp/pip-ephem-wheel-cache-6t10sf0s/wheels/0d/1d/98/9d7742bba16a500d029119527b82c4572bdb9d524fc955f470\n",
      "Successfully built katlas\n",
      "Installing collected packages: katlas\n",
      "Successfully installed katlas-0.0.1\n",
      "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!pip install -Uqq pyngrok\n",
    "!pip install .\n",
    "# !nbdev_preview"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "d8439e11-f2fb-43c8-a515-c051827443b8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Authtoken saved to configuration file: /root/.ngrok2/ngrok.yml                                      \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "t=2023-08-24T19:54:40+0000 lvl=warn msg=\"ngrok config file found at legacy location, move to XDG location\" xdg_path=/root/.config/ngrok/ngrok.yml legacy_path=/root/.ngrok2/ngrok.yml\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<NgrokTunnel: \"https://8d09-172-83-13-4.ngrok-free.app\" -> \"http://localhost:8000\">"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from pyngrok import ngrok\n",
    "!ngrok config add-authtoken 2TcZn9S1EnAWRznc1F3l07EhsDA_5DKkTteGNdVniRR6JMcAU\n",
    "ngrok_tunnel = ngrok.connect(8000)\n",
    "\n",
    "ngrok_tunnel\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1b57d9b2-986d-47d6-8e82-56500fa9e9b8",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[1m\u001b[34mPreparing to preview\u001b[39m\u001b[22m\n",
      "[ 1/20] tutorial_04a_prepare_dataset(protein only).ipynb\u001b[39m\u001b[22m\n",
      "[ 2/20] tutorial_03d_hierarchical_clustering-Copy1.ipynb\u001b[39m\u001b[22m\n",
      "[ 3/20] index.ipynb\u001b[39m\u001b[22m\n",
      "[ 4/20] tutorial_01_aa_feature.ipynb\u001b[39m\u001b[22m\n",
      "[ 5/20] tutorial_02_kinase_feature.ipynb\u001b[39m\u001b[22m\n",
      "[ 6/20] tutorial_04b_prepare_dataset.ipynb\u001b[39m\u001b[22m\n",
      "[ 7/20] 01_feature.ipynb\u001b[39m\u001b[22m\n",
      "[ 8/20] tutorial_05_train_ML_2.ipynb\u001b[39m\u001b[22m\n",
      "[ 9/20] 00_core.ipynb\u001b[39m\u001b[22m\n",
      "[10/20] 04_DL.ipynb\u001b[39m\u001b[22m\n",
      "[11/20] tutorial_03c_target_plot.ipynb\u001b[39m\u001b[22m\n",
      "[12/20] tutorial_05_train_ML.ipynb\u001b[39m\u001b[22m\n",
      "[13/20] tutorial_02b_kinase_EDA.ipynb\u001b[39m\u001b[22m\n",
      "[14/20] tutorial_03a_target_transform.ipynb\u001b[39m\u001b[22m\n",
      "[15/20] 04_DL-multi.ipynb\u001b[39m\u001b[22m\n",
      "[16/20] tutorial_03b_target_scale.ipynb\u001b[39m\u001b[22m\n",
      "[17/20] tutorial_05_train_protein_only.ipynb\u001b[39m\u001b[22m\n",
      "[18/20] 03_ML.ipynb\u001b[39m\u001b[22m\n",
      "[19/20] 02_plot.ipynb\u001b[39m\u001b[22m\n",
      "[20/20] tutorial_03d_hierarchical_clustering.ipynb\u001b[39m\u001b[22m\n",
      "\n",
      "\u001b[32mWatching files for changes\u001b[39m\n",
      "\u001b[32mBrowse at \u001b[39m\u001b[4m\u001b[32mhttp://localhost:8000/\u001b[39m\u001b[24m\n",
      "\u001b[32mGET: /\u001b[39m\n",
      "\u001b[32mGET: /\u001b[39m\n",
      "\u001b[32mGET: /\u001b[39m\n",
      "\u001b[32mGET: /feature.html\u001b[39m\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/quarto.py\", line 297, in _f\n",
      "    try: serve_drv.main(res)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/serve_drv.py\", line 22, in main\n",
      "    if src.suffix=='.ipynb': exec_nb(src, dst, x)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/serve_drv.py\", line 15, in exec_nb\n",
      "    nb = read_nb(src)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/execnb/nbio.py\", line 57, in read_nb\n",
      "    res = dict2nb(_read_json(path, encoding='utf-8'))\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/execnb/nbio.py\", line 16, in _read_json\n",
      "    return loads(Path(self).read_text(encoding=encoding, errors=errors))\n",
      "  File \"/usr/lib/python3.9/json/__init__.py\", line 346, in loads\n",
      "    return _default_decoder.decode(s)\n",
      "  File \"/usr/lib/python3.9/json/decoder.py\", line 337, in decode\n",
      "    obj, end = self.raw_decode(s, idx=_w(s, 0).end())\n",
      "  File \"/usr/lib/python3.9/json/decoder.py\", line 355, in raw_decode\n",
      "    raise JSONDecodeError(\"Expecting value\", s, err.value) from None\n",
      "json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/quarto.py\", line 297, in _f\n",
      "    try: serve_drv.main(res)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/serve_drv.py\", line 22, in main\n",
      "    if src.suffix=='.ipynb': exec_nb(src, dst, x)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/serve_drv.py\", line 15, in exec_nb\n",
      "    nb = read_nb(src)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/execnb/nbio.py\", line 57, in read_nb\n",
      "    res = dict2nb(_read_json(path, encoding='utf-8'))\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/execnb/nbio.py\", line 16, in _read_json\n",
      "    return loads(Path(self).read_text(encoding=encoding, errors=errors))\n",
      "  File \"/usr/lib/python3.9/json/__init__.py\", line 346, in loads\n",
      "    return _default_decoder.decode(s)\n",
      "  File \"/usr/lib/python3.9/json/decoder.py\", line 337, in decode\n",
      "    obj, end = self.raw_decode(s, idx=_w(s, 0).end())\n",
      "  File \"/usr/lib/python3.9/json/decoder.py\", line 355, in raw_decode\n",
      "    raise JSONDecodeError(\"Expecting value\", s, err.value) from None\n",
      "json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/quarto.py\", line 297, in _f\n",
      "    try: serve_drv.main(res)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/serve_drv.py\", line 22, in main\n",
      "    if src.suffix=='.ipynb': exec_nb(src, dst, x)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/serve_drv.py\", line 15, in exec_nb\n",
      "    nb = read_nb(src)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/execnb/nbio.py\", line 57, in read_nb\n",
      "    res = dict2nb(_read_json(path, encoding='utf-8'))\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/execnb/nbio.py\", line 16, in _read_json\n",
      "    return loads(Path(self).read_text(encoding=encoding, errors=errors))\n",
      "  File \"/usr/lib/python3.9/json/__init__.py\", line 346, in loads\n",
      "    return _default_decoder.decode(s)\n",
      "  File \"/usr/lib/python3.9/json/decoder.py\", line 337, in decode\n",
      "    obj, end = self.raw_decode(s, idx=_w(s, 0).end())\n",
      "  File \"/usr/lib/python3.9/json/decoder.py\", line 355, in raw_decode\n",
      "    raise JSONDecodeError(\"Expecting value\", s, err.value) from None\n",
      "json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/quarto.py\", line 297, in _f\n",
      "    try: serve_drv.main(res)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/serve_drv.py\", line 22, in main\n",
      "    if src.suffix=='.ipynb': exec_nb(src, dst, x)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/serve_drv.py\", line 15, in exec_nb\n",
      "    nb = read_nb(src)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/execnb/nbio.py\", line 57, in read_nb\n",
      "    res = dict2nb(_read_json(path, encoding='utf-8'))\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/execnb/nbio.py\", line 16, in _read_json\n",
      "    return loads(Path(self).read_text(encoding=encoding, errors=errors))\n",
      "  File \"/usr/lib/python3.9/json/__init__.py\", line 346, in loads\n",
      "    return _default_decoder.decode(s)\n",
      "  File \"/usr/lib/python3.9/json/decoder.py\", line 337, in decode\n",
      "    obj, end = self.raw_decode(s, idx=_w(s, 0).end())\n",
      "  File \"/usr/lib/python3.9/json/decoder.py\", line 355, in raw_decode\n",
      "    raise JSONDecodeError(\"Expecting value\", s, err.value) from None\n",
      "json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/quarto.py\", line 297, in _f\n",
      "    try: serve_drv.main(res)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/serve_drv.py\", line 22, in main\n",
      "    if src.suffix=='.ipynb': exec_nb(src, dst, x)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/serve_drv.py\", line 15, in exec_nb\n",
      "    nb = read_nb(src)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/execnb/nbio.py\", line 57, in read_nb\n",
      "    res = dict2nb(_read_json(path, encoding='utf-8'))\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/execnb/nbio.py\", line 16, in _read_json\n",
      "    return loads(Path(self).read_text(encoding=encoding, errors=errors))\n",
      "  File \"/usr/lib/python3.9/json/__init__.py\", line 346, in loads\n",
      "    return _default_decoder.decode(s)\n",
      "  File \"/usr/lib/python3.9/json/decoder.py\", line 337, in decode\n",
      "    obj, end = self.raw_decode(s, idx=_w(s, 0).end())\n",
      "  File \"/usr/lib/python3.9/json/decoder.py\", line 355, in raw_decode\n",
      "    raise JSONDecodeError(\"Expecting value\", s, err.value) from None\n",
      "json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/quarto.py\", line 297, in _f\n",
      "    try: serve_drv.main(res)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/serve_drv.py\", line 22, in main\n",
      "    if src.suffix=='.ipynb': exec_nb(src, dst, x)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/serve_drv.py\", line 15, in exec_nb\n",
      "    nb = read_nb(src)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/execnb/nbio.py\", line 57, in read_nb\n",
      "    res = dict2nb(_read_json(path, encoding='utf-8'))\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/execnb/nbio.py\", line 16, in _read_json\n",
      "    return loads(Path(self).read_text(encoding=encoding, errors=errors))\n",
      "  File \"/usr/lib/python3.9/json/__init__.py\", line 346, in loads\n",
      "    return _default_decoder.decode(s)\n",
      "  File \"/usr/lib/python3.9/json/decoder.py\", line 337, in decode\n",
      "    obj, end = self.raw_decode(s, idx=_w(s, 0).end())\n",
      "  File \"/usr/lib/python3.9/json/decoder.py\", line 355, in raw_decode\n",
      "    raise JSONDecodeError(\"Expecting value\", s, err.value) from None\n",
      "json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/quarto.py\", line 297, in _f\n",
      "    try: serve_drv.main(res)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/serve_drv.py\", line 22, in main\n",
      "    if src.suffix=='.ipynb': exec_nb(src, dst, x)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/serve_drv.py\", line 15, in exec_nb\n",
      "    nb = read_nb(src)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/execnb/nbio.py\", line 57, in read_nb\n",
      "    res = dict2nb(_read_json(path, encoding='utf-8'))\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/execnb/nbio.py\", line 16, in _read_json\n",
      "    return loads(Path(self).read_text(encoding=encoding, errors=errors))\n",
      "  File \"/usr/lib/python3.9/json/__init__.py\", line 346, in loads\n",
      "    return _default_decoder.decode(s)\n",
      "  File \"/usr/lib/python3.9/json/decoder.py\", line 337, in decode\n",
      "    obj, end = self.raw_decode(s, idx=_w(s, 0).end())\n",
      "  File \"/usr/lib/python3.9/json/decoder.py\", line 355, in raw_decode\n",
      "    raise JSONDecodeError(\"Expecting value\", s, err.value) from None\n",
      "json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/quarto.py\", line 297, in _f\n",
      "    try: serve_drv.main(res)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/serve_drv.py\", line 22, in main\n",
      "    if src.suffix=='.ipynb': exec_nb(src, dst, x)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/serve_drv.py\", line 15, in exec_nb\n",
      "    nb = read_nb(src)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/execnb/nbio.py\", line 57, in read_nb\n",
      "    res = dict2nb(_read_json(path, encoding='utf-8'))\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/execnb/nbio.py\", line 16, in _read_json\n",
      "    return loads(Path(self).read_text(encoding=encoding, errors=errors))\n",
      "  File \"/usr/lib/python3.9/json/__init__.py\", line 346, in loads\n",
      "    return _default_decoder.decode(s)\n",
      "  File \"/usr/lib/python3.9/json/decoder.py\", line 337, in decode\n",
      "    obj, end = self.raw_decode(s, idx=_w(s, 0).end())\n",
      "  File \"/usr/lib/python3.9/json/decoder.py\", line 355, in raw_decode\n",
      "    raise JSONDecodeError(\"Expecting value\", s, err.value) from None\n",
      "json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/quarto.py\", line 297, in _f\n",
      "    try: serve_drv.main(res)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/serve_drv.py\", line 22, in main\n",
      "    if src.suffix=='.ipynb': exec_nb(src, dst, x)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/serve_drv.py\", line 15, in exec_nb\n",
      "    nb = read_nb(src)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/execnb/nbio.py\", line 57, in read_nb\n",
      "    res = dict2nb(_read_json(path, encoding='utf-8'))\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/execnb/nbio.py\", line 16, in _read_json\n",
      "    return loads(Path(self).read_text(encoding=encoding, errors=errors))\n",
      "  File \"/usr/lib/python3.9/json/__init__.py\", line 346, in loads\n",
      "    return _default_decoder.decode(s)\n",
      "  File \"/usr/lib/python3.9/json/decoder.py\", line 337, in decode\n",
      "    obj, end = self.raw_decode(s, idx=_w(s, 0).end())\n",
      "  File \"/usr/lib/python3.9/json/decoder.py\", line 355, in raw_decode\n",
      "    raise JSONDecodeError(\"Expecting value\", s, err.value) from None\n",
      "json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/quarto.py\", line 297, in _f\n",
      "    try: serve_drv.main(res)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/serve_drv.py\", line 22, in main\n",
      "    if src.suffix=='.ipynb': exec_nb(src, dst, x)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/serve_drv.py\", line 15, in exec_nb\n",
      "    nb = read_nb(src)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/execnb/nbio.py\", line 57, in read_nb\n",
      "    res = dict2nb(_read_json(path, encoding='utf-8'))\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/execnb/nbio.py\", line 16, in _read_json\n",
      "    return loads(Path(self).read_text(encoding=encoding, errors=errors))\n",
      "  File \"/usr/lib/python3.9/json/__init__.py\", line 346, in loads\n",
      "    return _default_decoder.decode(s)\n",
      "  File \"/usr/lib/python3.9/json/decoder.py\", line 337, in decode\n",
      "    obj, end = self.raw_decode(s, idx=_w(s, 0).end())\n",
      "  File \"/usr/lib/python3.9/json/decoder.py\", line 355, in raw_decode\n",
      "    raise JSONDecodeError(\"Expecting value\", s, err.value) from None\n",
      "json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/quarto.py\", line 297, in _f\n",
      "    try: serve_drv.main(res)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/serve_drv.py\", line 22, in main\n",
      "    if src.suffix=='.ipynb': exec_nb(src, dst, x)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/serve_drv.py\", line 15, in exec_nb\n",
      "    nb = read_nb(src)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/execnb/nbio.py\", line 57, in read_nb\n",
      "    res = dict2nb(_read_json(path, encoding='utf-8'))\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/execnb/nbio.py\", line 16, in _read_json\n",
      "    return loads(Path(self).read_text(encoding=encoding, errors=errors))\n",
      "  File \"/usr/lib/python3.9/json/__init__.py\", line 346, in loads\n",
      "    return _default_decoder.decode(s)\n",
      "  File \"/usr/lib/python3.9/json/decoder.py\", line 337, in decode\n",
      "    obj, end = self.raw_decode(s, idx=_w(s, 0).end())\n",
      "  File \"/usr/lib/python3.9/json/decoder.py\", line 355, in raw_decode\n",
      "    raise JSONDecodeError(\"Expecting value\", s, err.value) from None\n",
      "json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/quarto.py\", line 297, in _f\n",
      "    try: serve_drv.main(res)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/serve_drv.py\", line 22, in main\n",
      "    if src.suffix=='.ipynb': exec_nb(src, dst, x)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/serve_drv.py\", line 15, in exec_nb\n",
      "    nb = read_nb(src)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/execnb/nbio.py\", line 57, in read_nb\n",
      "    res = dict2nb(_read_json(path, encoding='utf-8'))\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/execnb/nbio.py\", line 16, in _read_json\n",
      "    return loads(Path(self).read_text(encoding=encoding, errors=errors))\n",
      "  File \"/usr/lib/python3.9/json/__init__.py\", line 346, in loads\n",
      "    return _default_decoder.decode(s)\n",
      "  File \"/usr/lib/python3.9/json/decoder.py\", line 337, in decode\n",
      "    obj, end = self.raw_decode(s, idx=_w(s, 0).end())\n",
      "  File \"/usr/lib/python3.9/json/decoder.py\", line 355, in raw_decode\n",
      "    raise JSONDecodeError(\"Expecting value\", s, err.value) from None\n",
      "json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/quarto.py\", line 297, in _f\n",
      "    try: serve_drv.main(res)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/serve_drv.py\", line 22, in main\n",
      "    if src.suffix=='.ipynb': exec_nb(src, dst, x)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/serve_drv.py\", line 15, in exec_nb\n",
      "    nb = read_nb(src)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/execnb/nbio.py\", line 57, in read_nb\n",
      "    res = dict2nb(_read_json(path, encoding='utf-8'))\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/execnb/nbio.py\", line 16, in _read_json\n",
      "    return loads(Path(self).read_text(encoding=encoding, errors=errors))\n",
      "  File \"/usr/lib/python3.9/json/__init__.py\", line 346, in loads\n",
      "    return _default_decoder.decode(s)\n",
      "  File \"/usr/lib/python3.9/json/decoder.py\", line 337, in decode\n",
      "    obj, end = self.raw_decode(s, idx=_w(s, 0).end())\n",
      "  File \"/usr/lib/python3.9/json/decoder.py\", line 355, in raw_decode\n",
      "    raise JSONDecodeError(\"Expecting value\", s, err.value) from None\n",
      "json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/quarto.py\", line 297, in _f\n",
      "    try: serve_drv.main(res)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/serve_drv.py\", line 22, in main\n",
      "    if src.suffix=='.ipynb': exec_nb(src, dst, x)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/nbdev/serve_drv.py\", line 15, in exec_nb\n",
      "    nb = read_nb(src)\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/execnb/nbio.py\", line 57, in read_nb\n",
      "    res = dict2nb(_read_json(path, encoding='utf-8'))\n",
      "  File \"/usr/local/lib/python3.9/dist-packages/execnb/nbio.py\", line 16, in _read_json\n",
      "    return loads(Path(self).read_text(encoding=encoding, errors=errors))\n",
      "  File \"/usr/lib/python3.9/json/__init__.py\", line 346, in loads\n",
      "    return _default_decoder.decode(s)\n",
      "  File \"/usr/lib/python3.9/json/decoder.py\", line 337, in decode\n",
      "    obj, end = self.raw_decode(s, idx=_w(s, 0).end())\n",
      "  File \"/usr/lib/python3.9/json/decoder.py\", line 355, in raw_decode\n",
      "    raise JSONDecodeError(\"Expecting value\", s, err.value) from None\n",
      "json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)\n"
     ]
    }
   ],
   "source": [
    "!nbdev_preview --port 8000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4e17f7a0-fed2-49c6-9792-0565ff6fd7bc",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
